app/src/main/java/com/kii/gatewaysample/ui/fragments/GatewayServicesListFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.kii.gatewaysample.ui.fragments;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.DialogFragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.kii.gatewaysample.R;
+import com.kii.gatewaysample.model.UPnPService;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+/**
+ * List Gateway services found in LAN
+ */
+public class GatewayServicesListFragment extends DialogFragment {
+    public final static String GATEWAY_SERVICE="gateway_service";
+
+
+    @Bind(R.id.gatewayServicesListView)
+    ListView listView;
+
+    private GatewayServiceArrayAdapter adapter;
+    private UPnPService selectedService;
+    private UPnPService[] foundServices;
+
+    public GatewayServicesListFragment() {
+        // Required empty public constructor
+    }
+
+    public static GatewayServicesListFragment newFragment(UPnPService[] foundServices){
+        GatewayServicesListFragment fragment = new GatewayServicesListFragment();
+        fragment.foundServices = foundServices;
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.gateway_services_list_fragment, null);
+        getDialog().setTitle("Found Gateway Services");
+        ButterKnife.bind(this, view);
+        this.adapter = new GatewayServiceArrayAdapter(getActivity());
+        this.listView.setAdapter(adapter);
+        adapter.addAll(this.foundServices);
+        this.listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                String location = ((TextView) view.findViewById(R.id.gateway_description) ).getText().toString();
```

Comment:
`gateway_description ` is from item of list view, so can not bind with `ButterKnife`